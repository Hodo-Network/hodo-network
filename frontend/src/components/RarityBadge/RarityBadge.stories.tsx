import React from "react";
import { Story, Meta } from "@storybook/react";

import { RarityBadge, RarityBadgeProps } from "./RarityBadge";

export default {
  title: "Components/RarityBadge",
  component: RarityBadge,
  argTypes: {
    rarity: {
      options: ["common", "rare", "epic"],
      control: { type: "radio" },
    },
  },
} as Meta;

const Template: Story<RarityBadgeProps> = (args) => <RarityBadge {...args} />;

export const Common = Template.bind({});
Common.args = {
  rarity: "common",
};

export const Rare = Template.bind({});
Rare.args = {
  rarity: "rare",
};

export const Epic = Template.bind({});
Epic.args = {
  rarity: "epic",
};
