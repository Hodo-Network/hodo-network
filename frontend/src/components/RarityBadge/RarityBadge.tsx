import React from "react";
import Badge from "../Badge";
import { NFTRarity } from "../../typings/nft";

export const EpicBadge = {
  color: "red",
  label: NFTRarity.EPIC,
};

export const RareBadge = {
  color: "yellow",
  label: NFTRarity.RARE,
};

export const CommonBadge = {
  color: "blue",
  label: NFTRarity.COMMON,
};

export const BADGE_INFO = {
  [NFTRarity.RARE]: RareBadge,
  [NFTRarity.EPIC]: EpicBadge,
  [NFTRarity.COMMON]: CommonBadge,
};

export interface RarityBadgeProps {
  rarity: string;
}

export const RarityBadge: React.FC<RarityBadgeProps> = ({ rarity }) => {
  let info = CommonBadge;

  if (rarity === NFTRarity.RARE) {
    info = RareBadge;
  } else if (rarity === NFTRarity.EPIC) {
    info = EpicBadge;
  } else {
    info = CommonBadge;
  }

  return <Badge color={info.color} label={info.label} />;
};
