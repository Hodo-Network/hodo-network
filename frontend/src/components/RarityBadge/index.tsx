import React from "react";
import { NFTRarity } from "../../typings/nft";

export default function RarityBadge({
  rarity,
  className,
}: {
  rarity: string;
  className?: string;
}) {
  let badgeClass = "";

  switch (rarity) {
    case NFTRarity.RARE:
      badgeClass = "bg-red-100 text-red-800";
      break;

    case NFTRarity.EPIC:
      badgeClass = "bg-yellow-100 text-yellow-800";
      break;

    case NFTRarity.COMMON:
      badgeClass = "bg-blue-100 text-blue-800";
      break;

    default:
      badgeClass = "bg-gray-100 text-gray-800";
      break;
  }

  return (
    <span
      className={`inline-flex items-center px-2.5 py-1 rounded-md text-sm font-medium capitalize ${badgeClass} ${className}`}>
      {rarity}
    </span>
  );
}
