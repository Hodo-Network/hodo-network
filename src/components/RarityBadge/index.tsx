import React from "react";

export default function RarityBadge({
  rarity,
  className,
}: {
  rarity: string;
  className?: string;
}) {
  let badgeClass = "";

  switch (rarity) {
    case "Rare":
      badgeClass = "bg-red-100 text-red-800";
      break;

    case "Epic":
      badgeClass = "bg-yellow-100 text-yellow-800";
      break;

    case "Common":
      badgeClass = "bg-blue-100 text-blue-800";
      break;

    default:
      badgeClass = "bg-gray-100 text-gray-800";
      break;
  }

  return (
    <span
      className={`inline-flex items-center px-2.5 py-1 rounded-md text-sm font-medium ${badgeClass} ${className}`}>
      {rarity}
    </span>
  );
}
