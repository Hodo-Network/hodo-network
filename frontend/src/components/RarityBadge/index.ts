import { RarityBadge } from './RarityBadge';
export { RarityBadge }
