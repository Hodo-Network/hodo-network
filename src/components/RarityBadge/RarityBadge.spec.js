import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import RarityBadge from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<RarityBadge />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<RarityBadge />).toJSON();
  expect(tree).toMatchSnapshot();
});
