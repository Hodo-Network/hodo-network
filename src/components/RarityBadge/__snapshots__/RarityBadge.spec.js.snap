// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  className="inline-flex items-center px-2.5 py-1 rounded-md text-sm font-medium bg-gray-100 text-gray-800 undefined"
/>
`;
